Source/WebKit/WebProcess/GPU/graphics/RemoteRenderingBackendProxy.cpp

```diff
@@ -515,6 +514,24 @@ bool RemoteRenderingBackendProxy::isCached(const ImageBuffer& imageBuffer) const
     return false;
 }
 
+void RemoteRenderingBackendProxy::imageBufferSetDidPrepareForDisplay(WebKit::RemoteImageBufferSetIdentifier bufferSet, WebKit::ImageBufferSetPrepareBufferForDisplayOutputData outputData)
+{
+    m_background->putImageBufferSetPrepareForDisplayOutputData(bufferSet, WTFMove(outputData));
+}
+
+void RemoteRenderingBackendProxyBackground::putImageBufferSetPrepareForDisplayOutputData(RemoteImageBufferSetIdentifier bufferSet, ImageBufferSetPrepareBufferForDisplayOutputData outputData)
+{
+    m_imageBufferSetsOutputData.add({ bufferSet, 0 }, adoptRef(new ImageBufferSetPrepareBufferForDisplayOutputDataWrapper { WTFMove(outputData) }));
+}
+
+ImageBufferSetPrepareBufferForDisplayOutputData RemoteRenderingBackendProxyBackground::takeImageBufferSetPrepareForDisplayOutputData(RemoteImageBufferSetIdentifier bufferSet)
+{
+    auto handles = m_imageBufferSetsOutputData.take({ { bufferSet, 0 }, 0 }, IPC::Timeout::infinity());
```

Comment:
the threadsafeobjectheap is used here "incorrectly", not for purpose of what it is created. It's a lock+cv substitute for RemoteImageBuffersSetProxy.

Also: we should not wait for infinity. It may well be that the message never arrives.

instead of object heap, see the comment in the fence